### PR TITLE
tensor.AsType()

### PR DESCRIPTION
**Pull request by dcslin**

### CODE REVIEW

**Comment by dcslin** [comment]
Hi @joddiy , before i clean the code, please check if this works for you?
https://github.com/apache/singa/blob/41eca964faf7de14cea53c6ba465ce44e7cdfaa4/test/python/test_tensor.py#L340

**Comment by joddiy** [comment]
Hi, shicong, it works fine, thanks.

File: src/core/tensor/math_kernel.cu [hunk 1]
```diff
@@ -184,6 +184,20 @@ __global__ void KernelAbs(const size_t n, const float *in, float *out) {
   }
 }
 
+__global__ void KernelAsType2(const size_t n, const float *in, int *out) {
```
Comment (nudles) [inline src/core/tensor/math_kernel.cu]: can you give a better name?
like KernelCastInt2Float or KernelInt2Float

File: src/core/tensor/math_kernel.cu [hunk 2]
```diff
@@ -184,6 +184,20 @@ __global__ void KernelAbs(const size_t n, const float *in, float *out) {
...
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < n;
+       i += blockDim.x * gridDim.x) {
+    out[i] = in[i];
```
Comment (nudles) [inline src/core/tensor/math_kernel.cu]: do you need the cast operation like `out[i]=int(in[i])?

Comment (dcslin) [reply src/core/tensor/math_kernel.cu]: I think it is working as implicit casting now

File: src/core/tensor/math_kernel.cu (hunk 1 above)
Comment (dcslin) [reply src/core/tensor/math_kernel.cu]: Thank you, this is rectified.

**Comment by dcslin** [comment]
This is ready for merge

File: src/core/tensor/math_kernel.cu (hunk 2 above)
Comment (nudles) [reply src/core/tensor/math_kernel.cu]: better make it explicit.

Comment (dcslin) [reply src/core/tensor/math_kernel.cu]: thank you for the suggestion. it is explicited now.

**Pull request merged**